CMakeLists.txt

```diff
@@ -339,8 +342,18 @@ if(EMSCRIPTEN)
     add_compile_flag("-sDISABLE_EXCEPTION_CATCHING=0")
     add_link_flag("-sDISABLE_EXCEPTION_CATCHING=0")
   endif()
-  # make the tools immediately usable on Node.js
-  add_link_flag("-sNODERAWFS")
+  # make the tools immediately usable on Node.js, unless building for the browser
+  if (BUILD_FOR_BROWSER)
+    add_link_flag("-sENVIRONMENT=web,worker")
+    add_link_flag("-sINVOKE_RUN=0")
+    add_link_flag("-sMODULARIZE")
+    add_link_flag("-sEXPORT_ES6")
+    add_link_flag("-sFILESYSTEM")
+    add_link_flag("-sFORCE_FILESYSTEM")
+    add_link_flag("-sEXPORTED_RUNTIME_METHODS=run,callMain,FS")
```

Comment:
Can move this next to `INVOKE_RUN` as it is related (and the comment could cover both).

Reply:
Addressed in https://github.com/WebAssembly/binaryen/pull/6131/commits/5057ddd37d0a432f3203d2d5697bde10e16f64da.